source/credentials_utils.c

```diff
@@ -49,6 +71,80 @@ void aws_credentials_provider_invoke_shutdown_callback(struct aws_credentials_pr
     }
 }
 
+static bool s_parse_expiration_value_from_json_object(
+    struct aws_json_value *value,
+    const struct aws_parse_credentials_from_json_doc_options *options,
+    uint64_t *expiration_timepoint_in_seconds) {
+
+    if (value == NULL) {
+        AWS_LOGF_INFO(AWS_LS_AUTH_CREDENTIALS_PROVIDER, "No credentials Expiration field in Json document.");
+        return false;
+    }
+
+    struct aws_byte_cursor expiration_cursor = {
+        .ptr = NULL,
+        .len = 0,
+    };
+
+    switch (options->expiration_format) {
+        case AWS_PCEF_STRING_ISO_8601_DATE: {
+
+            if (!aws_json_value_is_string(value)) {
+                AWS_LOGF_INFO(
+                    AWS_LS_AUTH_CREDENTIALS_PROVIDER,
+                    "Expected string for type of credentials Expiration field in Json document.");
+                return false;
+            }
+
+            if (aws_json_value_get_string(value, &expiration_cursor)) {
+                AWS_LOGF_INFO(
```

Comment:
trivial: the only reason aws_json_value_get_string() will will raise an error is if it's not a string, so you could omit that `if (!aws_json_value_is_string(value))` check entirely